### PR TITLE
[BPK-3235] Add dark mode support to BpkNavigationBar

### PR DESCRIPTION
**Pull request by shaundon**

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -24,6 +24,9 @@
 - react-native-bpk-component-chip:
   - Added dark mode support.
 
+- react-native-bpk-component-navigation-bar:
+ - Added dark mode support.
```
Comment (georgegillams) [inline UNRELEASED.md]:
```suggestion
    - Added dark mode support.
```

File: UNRELEASED.md [hunk 2]
```diff
@@ -24,6 +24,9 @@
 - react-native-bpk-component-chip:
   - Added dark mode support.
 
+- react-native-bpk-component-navigation-bar:
```
Comment (georgegillams) [inline UNRELEASED.md]:
```suggestion
  - react-native-bpk-component-navigation-bar:
```

File: packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.ios.js [hunk 3]
```diff
@@ -57,13 +65,13 @@ const IOS_THEME_ATTRIBUTES = [
...
-const styles = StyleSheet.create({
+const dynamicStyles = BpkDynamicStyleSheet.create({
   barOuter: {
     flexDirection: 'column',
     paddingHorizontal: 8, // eslint-disable-line backpack/use-tokens
     width: '100%',
-    backgroundColor: colorSkyGrayTint07,
-    shadowColor: colorSkyGrayTint06,
+    backgroundColor: backgroundSecondaryColor,
+    shadowColor: { light: colorSkyGrayTint06, dark: colorBlackTint02 },
```
Comment (tiagoengel) [inline packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.ios.js]: I don't think we should have shadow in dark mode. This PR will allow us to provide `null` here https://github.com/Skyscanner/backpack-react-native/pull/296. 

Not a blocker I believe.

File: packages/react-native-bpk-component-navigation-bar/stories.js [hunk 4]
```diff
@@ -140,7 +140,7 @@ storiesOf('react-native-bpk-component-navigation-bar', module)
       leadingButton={backButton()}
       title="Backpack"
       subtitleView={
-        <BpkCard onPress={action('Card pressed')}>
+        <BpkCard onPress={action('Card pressed')} focused>
```
Comment (georgegillams) [inline packages/react-native-bpk-component-navigation-bar/stories.js]: What was the reason for this change?

File: storybook/storybook.js [hunk 5]
```diff
@@ -120,10 +119,6 @@ const BackgroundWrapper = ({ children }: { children: Node }) => {
       style={{
         backgroundColor,
         flex: 1,
-        marginTop: Platform.select({
```
Comment (georgegillams) [inline storybook/storybook.js]: What was the reason for this change?

File: packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBarTextButtonIOS.ios.js [hunk 6]
```diff
@@ -89,7 +91,12 @@ const BpkNavigationBarTextButtonIOS = (props: Props) => {
   const accessibilityStates = [];
   if (disabled) {
     accessibilityStates.push('disabled');
-    tintColorFinal = disabledTintColor || colorSkyGrayTint04;
+    tintColorFinal =
+      disabledTintColor ||
+      useBpkDynamicValue({
```
Comment (tiagoengel) [inline packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBarTextButtonIOS.ios.js]: Hooks should not be used inside conditionals. Can you move the call to the hook outside and just use the value here?

File: packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBarBackButtonIOS.ios.js [hunk 7]
```diff
@@ -53,7 +54,7 @@ const styles = StyleSheet.create({
 
 const BpkNavigationBarBackButtonIOS = (props: Props) => {
   const { title, showTitle, onPress, tintColor } = props;
-  const tintColorFinal = tintColor || colorSkyGray;
+  const tintColorFinal = tintColor || useBpkDynamicValue(textPrimaryColor);
```
Comment (tiagoengel) [inline packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBarBackButtonIOS.ios.js]: Hooks should not be used with conditionals, can we move the call to the hook a line up and just use the value here?

File: packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBarIconButtonIOS.ios.js [hunk 8]
```diff
@@ -69,7 +70,7 @@ const BpkNavigationBarIconButtonIOS = (props: Props) => {
   } = props;
   const tintColorFinal = disabled
     ? disabledTintColor || colorSkyGrayTint04
-    : tintColor || colorSkyGray;
+    : tintColor || useBpkDynamicValue(textPrimaryColor);
```
Comment (tiagoengel) [inline packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBarIconButtonIOS.ios.js]: same concern about conditionals

**Pull request merged**